### PR TITLE
Nmmsv/new gamma delta support

### PR DESCRIPTION
**Pull request by nmmsv10x**

### CODE REVIEW

File: enclone_args/src/read_json.rs [hunk 1]
```diff
@@ -247,14 +247,26 @@ fn parse_vector_entry_from_json(
...
+        if ctl.gen_opt.gamma_delta {
+            if ctl.gen_opt.trace_barcode == *barcode {
+                if !is_valid(&x, refdata, &ann, true, &mut log, Some(true)) {
+                    print!("{}", strme(&log));
+                    println!("invalid");
+                    return Ok(());
+                }
+            } else if !is_valid(&x, refdata, &ann, false, &mut log, Some(true)) {
+                return Ok(());
+            }
```
Comment (sreenathkrishnan) [inline enclone_args/src/read_json.rs:259]:
```suggestion
            if ctl.gen_opt.trace_barcode == *barcode {
                if !is_valid(&x, refdata, &ann, true, &mut log, Some(ctl.gen_opt.gamma_delta)) {
                    print!("{}", strme(&log));
                    println!("invalid");
                    return Ok(());
                }
            } else if !is_valid(&x, refdata, &ann, false, &mut log, Some(true)) {
                return Ok(());
            }
```

**Review by sreenathkrishnan — APPROVED** [review]
LGTM!

**Comment by DavidBJaffe** [comment]
I may have missed it, but we need to add at least one test for gamma-delta inside enclone.

**Comment by sreenathkrishnan** [comment]
> Please let's first create a branch off master. It does not make sense for the cellranger7.0 branch to have features that are not already on enclone master.

`cellranger7.0` will be merged back into master so there would not be any feature disparity. However, enclone master requires rust 1.56 whereas cellranger won't build on 1.56, so branching is necessary.

**Comment by DavidBJaffe** [comment]
Rust 1.57.0 should release today (Thursday) and cellranger should work with that.

**Comment by DavidBJaffe** [comment]
Noting that this appears to have falsely passed tests.  Somehow Jenkins did not run.

**Comment by evolvedmicrobe** [comment]
jenkins retest this please

**Review by DavidBJaffe — COMMENTED** [review]
Wonderful!  Things I'd like to see (ideally):
1. Keep executable size the same.  Not sure why it increased so much.
2. Add a test for the new functionality.  We can discuss.
3. Don't change prost specification unless actually needed.

**Comment by DavidBJaffe** [comment]
I did the experiment of modifying cellranger to point to my current enclone branch dj/264.  With that change, the code compiles (both `cargo b` and `cargo t --no-run`), so I don't understand why we need the `prost` change or why the executable size should grow a lot.

**Comment by nmmsv10x** [comment]
@sreenathkrishnan can you comment on the prost issue?
I think the problem that Sreenath found was that enclone_proto would not individually build without these additional features

**Comment by sreenathkrishnan** [comment]
On enclone master the following fails:
```
> cd enclone_proto/
> cargo check
```

#### Why does it fail?
Because the features `std` and `prost-derive` of prost are not enabled when you compile (`prost = { version = "0.9", default_features = false }` but those are actually required.

#### Why does it compile from the workspace root?
Due to [the way cargo resolve features in a workspace](https://doc.rust-lang.org/cargo/reference/resolver.html#features), cargo compilation is done with features `std` and `prost-derive` of prost. This is how the feature gets turned on:

- Multiple enclone workspace crates depend on `pretty_trace` crate
- `pretty_trace` [depends on prost](https://github.com/10XGenomics/rust-toolbox/blob/master/pretty_trace/Cargo.toml#L20) with the `protobuf` feature.
```
pprof = { version = "0.6", features = ["protobuf"] }
```
- `protobuf` feature of `pprof` enables `prost` as specified [in the Cargo.toml](https://github.com/tikv/pprof-rs/blob/master/Cargo.toml#L15) with default features
```
prost = { version = "0.9", optional = true }
```
- The features `std` and `prost-derive` are [enabled by default](https://github.com/tokio-rs/prost/blob/master/Cargo.toml#L43) in `prost`

Cellranger compiles enclone_proto successfully due to the same reason.

#### Why do we need to enable prost features?
- So that `enclone_proto` actually works as an independent crate
- Additionally it makes no changes to the `Cargo.lock` file in enclone, meaning it's not adding any new dependencies (as expected)

**Comment by nmmsv10x** [comment]
I also just ran the tests on the master branch branch of enclone, and the executable size grows by the same amount. I believe this means the growth is probably caused by toolchain update or something along those line.
`enclone executable is only allowed to change by 1%, but it has changed by 15.5%.`
`old size = 83371552, new size = 96271120`

**Comment by DavidBJaffe** [comment]
Executable size: yeah, I was not using the toolchain.  When I switch it on, the executable indeed gets much larger.  So yes the "change" is OK.  Why using the toolchain makes the
executable larger is an interesting question but out of scope of this PR,  ha ha.

prost: I'm confused but trust you guys.  I added a comment line to master.toml to keep
the change from accidentally getting reverted.  If you want to clarify that, feel free (but
not necessary).

Please feel free to merge!  We still need a test but that can come later if necessary.

**Pull request merged**